### PR TITLE
readability and consistency adjustments

### PR DESCRIPTION
**Pull request by FND**
As discussed elsewhere, here are a few suggestions for readability and consistency - separated into four commits so you can pick and choose:

1. whitespace
2. quotation marks
3. indentation
4. [bouncer pattern / early return](https://rikschennink.nl/thoughts/the-bouncer-pattern/)

Please see commit messages for details.

Let me emphasize that these are merely suggestions, mostly inconsequential and subject to personal preferences: Feel free to reject at will.

If you're interested, we might also add [EditorConfig support](https://prepitaph.org/snippets/editorconfig/) to ensure future consistency. I opted not to include that here for now.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -11,14 +11,15 @@ const trimMiddle = (str, length = 16, replaceString = "…") => {
...
+
+    let remainder = (length - replaceString.length) / 2;
+    let chunks = [...new Intl.Segmenter().segment(str)].map(x => x.segment);
+    return chunks.slice(0, Math.ceil(remainder)).join("") +
+            replaceString +
+            chunks.slice(-Math.floor(remainder)).join("");
 };
-export { trimMiddle };
\ No newline at end of file
+export { trimMiddle };
```
Comment (FND) [inline index.js:25]: To avoid duplication errors, we might add a small script to either

a) check that `index.js` and `clientside.js` are largely identical, or

b) generate `index.js` from `clientside.js`

Not sure that's worth the trouble though.

File: test/script.js [hunk 2]
```diff
@@ -1,6 +1,5 @@
 // test/script.js
 
-const teststring = require('trimmiddle')
-
-console.log(teststring('This is a long string that will be trimmed'));
+const teststring = require("trimmiddle");
 
+console.log(teststring("This is a long string that will be trimmed"));
```
Comment (FND) [inline test/script.js:5]: I wasn't sure how you execute this script, so I created a local variant `script.cjs` and ran it through Node:

```javascript
const { trimMiddle } = require("../index.js");

console.log(trimMiddle("This is a long string that will be trimmed"));
```

(Note that both file extension and import differ from the original.)

Might be worth documenting the process in `README.md`? Or perhaps add a few simple tests using [Node's built-in test runner](https://nodejs.org/api/test.html#running-tests-from-the-command-line)?

File: clientside.js [hunk 3]
```diff
@@ -11,13 +11,14 @@ const trimMiddle = (str, length = 16, replaceString = "…") => {
...
-        chunks.slice(-Math.floor(remainder)).join('');
+
+    if (str.length <= length) {
+        return str;
     }
-};
\ No newline at end of file
+
+    let remainder = (length - replaceString.length) / 2;
+    let chunks = [...new Intl.Segmenter().segment(str)].map(x => x.segment);
```
Comment (FND) [inline clientside.js:20]: Minor performance-optimization potential: This allocates multiple arrays, which _might_ become a significant factor in garbage collection if this function is invoked often enough. (Seems unlikely, but who knows?)

An alternative approach might look like this:

```suggestion
    let chunks = [];
    for (let chunk of new Intl.Segmenter().segment(str)) {
        chunks.push(chunk);
    }
```

File: index.js (hunk 1 above)
Comment (codepo8) [reply index.js:25]: I wondered about that, too, or make it a github action. Maybe when a lot of changes come in.

File: test/script.js (hunk 2 above)
Comment (codepo8) [reply test/script.js:5]: test suite is a thing to work on, already there was a report of an edge case bug.

File: clientside.js (hunk 3 above)
Comment (codepo8) [reply clientside.js:20]: Good catch, often a map seems clever, but can cause issues.
